src/coreComponents/dataRepository/BufferOps_inline.hpp

```diff
@@ -290,18 +404,33 @@ localIndex PackByIndex( buffer_unit_type * & buffer,
   return sizeOfPackedChars;
 }
 
+template< bool DO_PACKING, typename MAP_TYPE, typename T_INDICES >
+typename std::enable_if< is_host_packable_map_by_index< MAP_TYPE >, localIndex >::type
+PackDataByIndex( buffer_unit_type * & buffer,
+                 MAP_TYPE const & var,
+                 T_INDICES const & indices )
+{
+  localIndex sizeOfPackedChars = 0;
+  for( typename MAP_TYPE::const_iterator i = var.begin(); i != var.end(); ++i )
+  {
+    sizeOfPackedChars += PackData< DO_PACKING >( buffer, i->first );
+    sizeOfPackedChars += PackDataByIndex< DO_PACKING >( buffer, i->second, indices );
+  }
+  return sizeOfPackedChars;
+}
+
 template< bool DO_PACKING, typename MAP_TYPE, typename T_INDICES >
 typename std::enable_if< is_host_packable_map_by_index< MAP_TYPE >, localIndex >::type
 PackByIndex( buffer_unit_type * & buffer,
              MAP_TYPE const & var,
-             T_INDICES const & packIndices )
+             T_INDICES const & indices )
 {
   const typename MAP_TYPE::size_type length = var.size();
   localIndex sizeOfPackedChars = Pack< DO_PACKING >( buffer, length );
   for( typename MAP_TYPE::const_iterator i = var.begin(); i != var.end(); ++i )
   {
     sizeOfPackedChars += Pack< DO_PACKING >( buffer, i->first );
-    sizeOfPackedChars += PackByIndex< DO_PACKING >( buffer, i->second, packIndices );
+    sizeOfPackedChars += PackByIndex< DO_PACKING >( buffer, i->second, indices );
```

Comment:
@wrtobin it seems like you do not reuse the PackDataByIndex function you created just above by extracting the for loop

Reply:
The refactoring to internally reuse the `data` versions of these functions (packing without any metadata) from within the standard `pack` version isn't straightforward. 

The `data` versions are used in the `Wrapper` when we're packing without embedded metadata, which is the case when packing for `TimeHistory`. 

Really all that matters is that each pair of `pack/unpack` functions with the same variations work together. 

Although we don't implement `unpackDataByIndex`, since it isn't possible to unpack those structures without the embedded metadata in practice.

Reply:
it seems that something is going wrong with test testRestartBasic, when conduit is packing the nodes

Reply:
@wrtobin  is the refactoring necessary ? the tests are passing locally on my machine without it

Reply:
Not having a `packDataByIndex` which doesn't include metadata in the serialization of the underlying data will cause the `TimeHistory` output to be corrupted when packing the data on host. 

The `TimeHistory` output requires the serialization only contain data since it buffers / writes out the entire serialized buffer of data, without inspecting / operating on it (other than a size check IIRC).

I have to look through the tests. Possibly a call is being made with the old signature, or when I modified the internal calls I mixed up a `true/false` since the old signature had two adjacent bool arguments, one for device-packing, and one to include metadata.